pandas/io/excel/_openpyxl.py

```diff
@@ -451,3 +462,277 @@ def write_cells(self, cells, sheet_name=None, startrow=0, startcol=0,
                             for k, v in style_kwargs.items():
                                 setattr(xcell, k, v)
+
+
+class _OpenpyxlReader(_BaseExcelReader):
+
+    def __init__(self, filepath_or_buffer):
+        """Reader using openpyxl engine.
+
+        Parameters
+        ----------
+        filepath_or_buffer : string, path object or Workbook
+            Object to be parsed.
+        """
+        err_msg = "Install openpyxl >= 2.4.0 for Excel with Openpyxl support"
+
+        try:
+            import openpyxl
+        except ImportError:
+            raise ImportError(err_msg)
+        else:
+            if openpyxl.__version__ < LooseVersion('2.4.0'):
+                raise ImportError(err_msg +
+                                  ". Current version " + openpyxl.__version__)
+        super().__init__(filepath_or_buffer)
+
+    @property
+    def _workbook_class(self):
+        from openpyxl import Workbook
+        return Workbook
+
+    def load_workbook(self, filepath_or_buffer):
+        from openpyxl import load_workbook
+        return load_workbook(filepath_or_buffer, data_only=True)
+
+    @property
+    def sheet_names(self):
+        return self.book.sheetnames
+
+    @staticmethod
+    def _handle_usecols(frame, usecols):
+        column_names = frame.columns.values
+        if usecols:
+            _validate_usecols_arg(usecols)
+            usecols = sorted(usecols)
+            if any(isinstance(i, str) for i in usecols):
+                _validate_usecols_names(usecols, column_names)
+                frame = frame[usecols]
+            else:
+                frame = frame.iloc[:, usecols]
+        return frame
+
+    def _handle_sheet_name(self, sheet_name):
+        """Handle the sheet_name keyword."""
+        # Keep sheetname to maintain backwards compatibility.
+        if isinstance(sheet_name, list):
+            sheets = sheet_name
+        elif sheet_name is None:
+            sheets = self.sheet_names
+        else:
+            sheets = [sheet_name]
+        return sheets
+
+    @staticmethod
+    def _handle_header_keywords(data, header, skiprows, index_col):
+        """Handle keywords relating to header parsing."""
+        # forward fill and pull out names for MultiIndex column
+        header_names = None
+        if header is not None and is_list_like(header):
+            header_names = []
+            control_row = [True] * len(data[0])
+
+            for row in header:
+                if is_integer(skiprows):
+                    row += skiprows
+
+                data[row], control_row = _fill_mi_header(data[row],
+                                                         control_row)
+
+                if index_col is not None:
+                    header_name, _ = _pop_header_name(data[row], index_col)
+                    header_names.append(header_name)
+        return header_names
+
+    @staticmethod
+    def _handle_convert_float(series, convert_float):
```

Comment:
Looks like this is causing test failures because you are getting more than just Series objects passed in as first argument

Reply:
Yep, something is wrong there. Seems to be related to duplicate column names, and while going down that rabbithole I might have stumbled on a bug. Don't have time to go through this in depth, but can you confirm this is not working while it should:

```
from pandas import DataFrame
nan = float('nan')
data = [[None, 'a', 'b', 'a', 'b'], [None, None, None, None, None], [0, 1, 2.5, 1, True], ]
column_names = [None, 'foo', nan, 'bar', nan]
frame = DataFrame(data, columns=column_names)
frame.iloc[:,2] = frame.iloc[:,2]
```